File: Logic_Building_Problems/4_sum_of_squares_of_naturals.py
def getSumOfSquares(number):
    return sum([n**2 for n in range(1, number+1)])

def getSumOfSquaresWithFormula(number):
    return (number * (number + 1) * (2 * number + 1)) // 6

if __name__ == '__main__':
    input_number = int(input())
    # print(getSumOfSquares(number=input_number))
    print(getSumOfSquaresWithFormula(number=input_number))

File: Logic_Building_Problems/4_sum_of_squares_of_naturals.cpp
#include <iostream>
using namespace std;

int getSumOfSquareWithLoop(int n) {
    int sum = 0;
    for(int i=0; i<=n; i++) {
        sum += i*i;
    }
    return sum;
}

int getSumOfSquareWithFormula(int n) {
    return (n * (n + 1) * (2 * n + 1)) / 6;
}


int main() {
    int input_number = 0;
    cin >> input_number;

    // cout << getSumOfSquareWithLoop(input_number) << endl;
    cout << getSumOfSquareWithFormula(input_number) << endl;
}